src/NuGet.Core/NuGet.Commands/RestoreCommand/CompatibilityChecker.cs

```diff
@@ -243,17 +243,21 @@ private static RestoreLogMessage GetErrorMessage(NuGetLogCode logCode, Compatibi
             {
                 foreach (var group in contentItems.FindItemGroups(pattern))
                 {
-                    object tfmObj = null;
-                    object ridObj = null;
-                    group.Properties.TryGetValue(ManagedCodeConventions.PropertyNames.RuntimeIdentifier, out ridObj);
-                    group.Properties.TryGetValue(ManagedCodeConventions.PropertyNames.TargetFrameworkMoniker, out tfmObj);
+                    // lib/net45/subfolder/a.dll will be returned as a group with zero items since sub
+                    // folders are not allowed. Completely empty groups are not compatible, a group with
+                    // _._ would contain _._ as an item.
+                    if (group.Items.Count > 0)
+                    {
+                        group.Properties.TryGetValue(ManagedCodeConventions.PropertyNames.RuntimeIdentifier, out var ridObj);
+                        group.Properties.TryGetValue(ManagedCodeConventions.PropertyNames.TargetFrameworkMoniker, out var tfmObj);
 
-                    var tfm = tfmObj as NuGetFramework;
+                        var tfm = tfmObj as NuGetFramework;
 
-                    // RID specific items should be ignored here since they are only used in the runtime assem check
-                    if (ridObj == null && tfm?.IsSpecificFramework == true)
-                    {
-                        available.Add(tfm);
+                        // RID specific items should be ignored here since they are only used in the runtime assem check
```

Comment:
*runtime assembly check?

Reply:
I've updated the existing comment to make this more clear.
In the interest of science: reloading this in VS, modifying it, and re-verify the build took 09:14.79